styx-scheduler-service/src/test/java/com/spotify/styx/SystemTest.java

```diff
@@ -176,6 +178,40 @@ public void shouldRunCustomWorkflowSchedule() throws Exception {
   }
   }
 
+  public static class ShouldRunFlyteWorkflowToCompletion extends SystemTest {
+    @Test
+    public void shouldRunFlyteWorkflowToCompletion() throws Exception {
+      Workflow wf = Workflow.create(
+          "comp",
+          WorkflowConfiguration.builder()
+              .id("flyte-wf")
+              .schedule(Schedule.parse("45 12 * * *"))
+              .flyteExecConf(FLYTE_EXEC_CONF)
+              .build());
+
+      givenTheTimeIs("2016-03-14T15:30:00Z");
+      givenWorkflow(wf);
+      givenWorkflowEnabledStateIs(wf, true);
+      givenNextNaturalTrigger(wf, "2016-03-14T12:45:00Z");
```

Comment:
shouldn't this timestamp be after the givenTheTimeIs ?

Reply:
or is this why you run the tickTriggerManager(); on line 200 ?

Reply:
The instant will only run be triggered if it is before current time.
https://github.com/spotify/styx/blob/master/styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java#L130

Reply:
👍